joint_trajectory_controller/include/joint_trajectory_controller/joint_trajectory_controller.h

```diff
@@ -246,6 +246,12 @@ class JointTrajectoryController : public controller_interface::Controller<Hardwa
    */
   void setHoldPosition(const ros::Time& time, RealtimeGoalHandlePtr gh=RealtimeGoalHandlePtr());
 
+private:
+  virtual bool checkStates(const std::vector<double>& old_desired_position,
```

Comment:
I think it is important to document that these functions have to be "realtime-safe" since they are called from `update()`.

Reply:
Done.